webapp/components/wpt-metadata.js

```diff
@@ -170,13 +184,39 @@ class WPTMetadata extends PathInfo(LoadingState(PolymerElement)) {
     url.searchParams.set('includeTestLevel', true);
     url.searchParams.set('products', productVal.join(','));
     this.load(
-      window.fetch(url).then(r => r.json()).then(metadata => {
-        this.metadata = metadata;
+      window.fetch(url).then(r => r.json()).then(mergedMetadata => {
+        this.mergedMetadata = mergedMetadata;
       })
     );
   }
 
-  computTestResultSet(searchResults) {
+  // loadPendingMetadata is called when wpt-metadata.js is initialized
+  // through constructor() or when users triage new metadata, unlike loadMergedMetadata().
```

Comment:
This feels like it could cause triaged metadata UI to disappear when in a triaging session? E.g.:

```
1. Triage test foo.html. A pending PR is created, it is added to /api/metadata/pending, and it shows up in the UI
2. Continuing triaging. In the background, the created PR is merged because its good.
3. The searchcache periodic sync runs, and cleans up the pending PR metadata
4. The user triages another test, bar.html. The metadata icon for foo.html disappears ?!
```

Is this correct? If so, should we be thinking about refreshing `/api/metadata` as well?

Reply:
For step 4, When user triages another test, webapp will force fetch the latest copy of the wpt-metadata repo, thus icons will remain. However, if user refreshes the page, there is a time period where icons will disappear. Refreshing `/api/metadata` won't help because the in-memory copy could still be old. Let me think about it more...

Reply:
To solve this issue, I will only remove a pending PR from /api/pending/metadata 10 minutes after it is merged

Reply:
Will send out a fix in a separate PR

Reply:
https://github.com/web-platform-tests/wpt.fyi/pull/2515